src/nvim/strings.c

```diff
@@ -45,6 +45,41 @@
 #include "nvim/os/os.h"
 #include "nvim/os/shell.h"
 
+/// Compares two strings for equality.
+bool strequal(const char *a, const char *b)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strcmp(a, b) == 0;
+}
+
+/// Compares two sized strings for equality.
+bool strnequal(const char *a, const char *b, size_t size)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strncmp(a, b, size) == 0;
+}
+
+/// Compares two strings for equality ignoring differences in case.
+bool striequal(const char *a, const char *b)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strcasecmp(a, b) == 0;
```

Comment:
> @aktau And third condition should not be a error, it should be vim_stricmp.

`strcasecmp` is supported on Linux, FreeBSD, OpenBSD, DragonFlyBSD and Illumos (checked by finding the relevant docs on Google). `_stricmp` is supported on Windows. These are all of our supported platforms and then some. I would actually advocate that we remove `vim_stricmp()` from our codebase.

If anyone bumps into the `#error`, we'll get an issue and we can either re-instate `vim_stricmp()` or find the case insensitive comparison function for that platform instead.